cmd/ruler/main.go

```diff
@@ -50,13 +50,14 @@ func main() {
 
 	util.InitLogger(&serverConfig)
 
-	storageClient, err := storage.NewStorageClient(storageConfig, schemaConfig)
+	storageOpts, err := storage.Clients(storageConfig, schemaConfig)
 	if err != nil {
 		level.Error(util.Logger).Log("msg", "error initializing storage client", "err", err)
 		os.Exit(1)
 	}
+	schemaOpts := chunk.SchemaOpts(chunkStoreConfig, schemaConfig)
```

Comment:
could this line be done inside `chunk.NewStore` since it looks like it is the same everywhere? Or is there a case you would want to use a NewStore with schemaOpts not generated like this?